R/LearnerSurvBlackboost.R

```diff
@@ -0,0 +1,195 @@
+          ParamLgl$new(id = "multiway", default = FALSE, tags = "train"),
+          ParamInt$new(id = "splittry", default = 2L, lower = 1L, tags = "train"),
+          ParamLgl$new(id = "intersplit", default = FALSE, tags = "train"),
+          ParamLgl$new(id = "majority", default = FALSE, tags = "train"),
+          ParamLgl$new(id = "caseweights", default = TRUE, tags = "train"),
+          ParamDbl$new(
+            id = "sigma", default = 0.1, lower = 0, upper = 1,
+            tags = c("train", "cindex")),
+          ParamUty$new(id = "ipcw", default = 1, tags = c("train", "cindex")),
+          ParamUty$new(id = "na.action", default = na.omit, tags = "train")
+        )
+      )
+
+      ps$values = list(family = "coxph")
+      ps$add_dep("sigma", "family", CondEqual$new("cindex"))
+      ps$add_dep("ipcw", "family", CondEqual$new("cindex"))
+
+      super$initialize(
+        id = "surv.blackboost",
+        param_set = ps,
+        feature_types = c("integer", "numeric", "factor"),
+        predict_types = c("distr", "crank", "lp", "response"),
+        properties = c("weights"),
+        packages = "mboost"
+      )
+    }
+  ),
+
+  private = list(
+    .train = function(task) {
+
+      pars = self$param_set$get_values(tags = "train")
+
+      if ("weights" %in% task$properties) {
+        pars$weights = task$weights$weight
+      }
+
+      # mboost control
+      # Save control settings and return on exit
+      saved_ctrl = mboost::boost_control()
+      is_ctrl_pars = (names(pars) %in% names(saved_ctrl))
+      # ensure only relevant pars passed to fitted model
+      if (any(is_ctrl_pars)) {
+        pars$control = do.call(mboost::boost_control, pars[is_ctrl_pars])
+        pars = pars[!is_ctrl_pars]
+      }
+
+      # GenzBretz control
+      # Save control settings and return on exit
+      saved_ctrl = mvtnorm::GenzBretz()
+      is_ctrl_pars = (names(pars) %in% names(saved_ctrl))
+      # ensure only relevant pars passed to fitted model
+      if (any(is_ctrl_pars)) {
+        pars$pargs = do.call(mvtnorm::GenzBretz, pars[is_ctrl_pars])
+        pars = pars[!is_ctrl_pars]
+      }
+
+      # ctree control
+      # Save control settings and return on exit
+      saved_ctrl = partykit::ctree_control()
+      is_ctrl_pars = (names(pars) %in% names(saved_ctrl))
+      # ensure only relevant pars passed to fitted model
+      if (any(is_ctrl_pars)) {
+        pars$tree_controls = do.call(partykit::ctree_control, pars[is_ctrl_pars])
+        pars = pars[!is_ctrl_pars]
+      }
+
+      # if ("weights" %in% task$properties)
+      #   pars$weights = task$weights$weight
+
+      family = switch(pars$family,
+        coxph = mboost::CoxPH(),
+        weibull = mlr3misc::invoke(mboost::Weibull,
+          .args = self$param_set$get_values(tags = "aft")),
+        loglog = mlr3misc::invoke(mboost::Loglog,
+          .args = self$param_set$get_values(tags = "aft")),
```

Comment:
We decided to not use tags for splitting up the parameter set. You can use `pars[which(names(pars) %in% formalArgs(mboost::Loglog))]` to get the parameter subset for the `mboost::Loglog` function.

Reply:
sorry for some reason never got a notification for this, removed tags as requested